airflow/__init__.py

```diff
@@ -74,11 +74,13 @@ def __getattr__(name):
 if not settings.LAZY_LOAD_PROVIDERS:
     from airflow import providers_manager
 
-    providers_manager.ProvidersManager().initialize_providers_manager()
+    manager = providers_manager.ProvidersManager()
+    manager.initialize_providers_list()
```

Comment:
Rather than having to "remember" call this everywhere we need it, I think it would be best if init hooks and init extra_links called this itself.

Reply:
right.